modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/LogicalScanConverterRule.java

```diff
@@ -71,7 +106,23 @@
                 RelMetadataQuery mq,
                 IgniteLogicalTableScan rel
             ) {
-                RelTraitSet traits = rel.getTraitSet().replace(IgniteConvention.INSTANCE);
+                RelOptCluster cluster = rel.getCluster();
+                IgniteTable table = rel.getTable().unwrap(IgniteTable.class);
+
+                RelDistribution distribution = table.distribution();
+                if (rel.requiredColumns() != null) {
+                    Mappings.TargetMapping mapping = createMapping(
+                        rel.projects(),
+                        rel.requiredColumns(),
+                        table.getRowType(cluster.getTypeFactory()).getFieldCount()
+                    );
+
+                    distribution = distribution.apply(mapping);
+                }
+
+                RelTraitSet traits = cluster.traitSetOf(IgniteConvention.INSTANCE)
```

Comment:
uhu ! u found ANY distribution source ?)

Reply:
before this patch the traits partially were set for logical rels and transitively were applied to the physical rels. Now the traits are created for physical rels only